floris/tools/optimization/layout_optimization/layout_optimization_base.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2022 NREL
+
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+# See https://floris.readthedocs.io for documentation
+
+import numpy as np
+import matplotlib.pyplot as plt
+from shapely.geometry import Polygon, LineString
+
+from ....logging_manager import LoggerBase
+
+class LayoutOptimization(LoggerBase):
+    def __init__(self, fi, boundaries, min_dist=None, freq=None):
+        self.fi = fi
+        self.boundaries = boundaries
+
+        self.boundary_polygon = Polygon(self.boundaries)
+        self.boundary_line = LineString(self.boundaries)
+
+        self.xmin = np.min([tup[0] for tup in boundaries])
+        self.xmax = np.max([tup[0] for tup in boundaries])
+        self.ymin = np.min([tup[1] for tup in boundaries])
+        self.ymax = np.max([tup[1] for tup in boundaries])
+        self.x0 = self._norm(self.fi.layout_x, self.xmin, self.xmax)
```

Comment:
How about having a separate variable for the initial conditions (turbine locations)? Something like `x0`, which would default to `layout_array`? This way, one can easily try different initial conditions without having to manipulate the `floris` object.

Reply:
I spent some time thinking about this, and decided to remove the option to specify x0 altogether and have it just be taken from the floris interface. If we allowed the users to change the initial conditions of the optimization directly within the object, we would need a  `reinitialize_opt` function, which then we get in the same issue of  needing to make sure everything gets updated properly (positions get changed in the `fi` object, the initial AEP is re-calculated, etc.). Also, it doesn't require the user to know whether they should input the `x0` as normalized or unnormalized, and the different optimization packages (Scipy vs pyOptSparse) require `x0` in different formats anyway. We could add this as a feature in the future, but it would require a significnat overhaul of the optimization classes and would need some more thought.